src/core/processing/qgsprocessingparameters.h

```diff
@@ -1849,6 +1849,58 @@ class CORE_EXPORT QgsProcessingParameterFolderDestination : public QgsProcessing
 
 };
 
+/**
+ * \class QgsProcessingParameterBand
+ * \ingroup core
+ * A raster band parameter for Processing algorithms.
+  * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsProcessingParameterBand : public QgsProcessingParameterDefinition
+{
+  public:
+
+    /**
+     * Constructor for QgsProcessingParameterBand.
+     */
+    QgsProcessingParameterBand( const QString &name, const QString &description = QString(), const QVariant &defaultValue = QVariant(),
+                                const QString &parentLayerParameterName = QString(),
```

Comment:
IMHO better to rename getter to `parentLayerParameterName()` as it returns name of the parameter, not layer parameter itself. Do you want me to take care about it?